llama_index/indices/document_summary/retrievers.py

```diff
@@ -74,6 +76,56 @@ def default_parse_choice_select_answer_fn(
     return answer_nums, answer_relevances
 
 
+class DocumentSummaryRetriever(BaseRetriever):
```

Comment:
To make sure I understand, this Retriever is added to avoid explicitly building the index?

Reply:
Yea I agree with all the above. Having it in `indices/list/retrievers.py` makes more sense than here. So something like `ListIndexLLMRetriever` in that file?

Any thoughts on whether to duplicate vs share the `format_node..` and `parse_choice_..` pieces?

Reply:
yep! `ListIndexLLMRetriever` sounds great 

could try sharing the `format_node_batch_fn` and `parse_choice` functions into `llama_index/indices/utils.py`? you can put it there and we can refactor later